packages/pages/src/components/analytics/Analytics.ts

```diff
@@ -0,0 +1,125 @@
+  // TODO: create event queue for things to fire when optIn is called if we are not optedIn
+  private _analyticsReporter: PagesAnalyticsService|undefined;
+  constructor(
+    private templateData?: TemplateProps,
+    requireOptIn?: boolean|undefined,
+  ) {
+    this.requireOptIn = requireOptIn;
+    this._optedIn = !requireOptIn;
+    this.makeReporter();
+    this.pageView();
+  }
+
+  calculatePageType(): EntityPage|StaticPage {
+    if (this.templateData === undefined) {
+      throw new Error('template data is required');
+    }
+
+    const isStaticPage = !!this.templateData?.document?.__?.staticPage;
+    const isEntityPage = !!this.templateData?.document?.__?.entityPageSet;
+
+    let pageType: EntityPage|StaticPage;
+
+    if(isStaticPage) {
+      pageType = {
+        name: 'static',
+        staticPageId: this.templateData.document.__.name as string,
+      }
+    } else if (isEntityPage) {
+      pageType = {
+        name: 'entity',
+        pageSetId: this.templateData.document.__.name as string,
+        id: this.templateData.document.uid as number,
+      }
+    } else {
+      throw new Error('invalid document type');
+    }
+
+    return pageType;
+  }
+
+  makeReporter() {
+    if (getRuntime().name !== 'browser') return;
+    if (!this._optedIn) return;
+    if (!this.templateData) {
+      throw new Error('Attempted to instantiate Analytics without required template details');
+    }
+    this._analyticsReporter = providePagesAnalytics({
+      businessId: this.templateData.document.businessId as number,
+      pageType: this.calculatePageType(),
+      pageUrl: window.location.href,
+      production: false,
+      referrer: document.referrer,
+      siteId: this.templateData.document.siteId as number,
+    });
+  }
+
+  private canTrack() :boolean {
+    return getRuntime().name ==='browser'
+      &&  this._optedIn
+      && !!this._analyticsReporter;
+  }
+
+  enableConversionTracking(): void {
+    if (this.canTrack()) {
+      this._cookieManager = new CookieManager();
+      this._analyticsReporter?.setConversionTrackingEnabled(
+        true,
+        this._cookieManager.setAndGetYextCookie(),
+      );
+    }
+  }
+
+  identify(visitor: Visitor): void {
+    if (this.canTrack()) {
+      this._analyticsReporter?.setVisitor(visitor);
+    }
+  }
+
+  optIn(): void {
+    this._optedIn = true;
+  }
+
+  async pageView(): Promise<void> {
+    if(!this.canTrack()) return Promise.resolve(undefined);
+    await this._analyticsReporter?.pageView();
+  }
+
+  pushScope(name: string): void {
+    this._scopes.push(Analytics.slugify(name));
+  }
+
+  private static slugify(str: string): string {
+    return str
+      .toLowerCase()
+      .trim()
+      .replace(/[^\w\s-]/g, '')
+      .replace(/[\s_-]+/g, '')
+      .replace(/^-+|-+$/g, '');
+  }
+
+  async track(eventName: string, conversionData?: ConversionDetails): Promise<void> {
+    if(!this.canTrack()) return Promise.resolve(undefined);
```

Comment:
Use braces, same throughout

Reply:
probably should add that to the eslint config, if it's a style standard.